sympy/printing/printer.py

```diff
@@ -326,7 +326,7 @@ def _print(self, expr, **kwargs):
             for cls in classes:
                 printmethod = '_print_' + cls.__name__
                 if hasattr(self, printmethod):
-                    return getattr(self, printmethod)(expr, **kwargs)
```

Comment:
@moorepants, why this change is failing in some checks? Have I done it falsely?

Reply:
`**kwargs` must be there since they control the exact behavior of the printing. So this must be restored.

Probably the failing tests are due to this as it changes the printing of (more or less) *all* functions. 

You should approach the method `_print_SingularityFunction` to introduce the changes.